MigrationExecutorApp/ChangeFeedProcessortHost.cs

```diff
@@ -108,92 +102,221 @@ public async Task CloseAsync()
             changeFeedProcessor = null;
         }
 
-        public async Task<IChangeFeedProcessor> RunChangeFeedHostAsync()
+        public async Task<ChangeFeedProcessor> RunChangeFeedHostAsync()
         {
             string hostName = Guid.NewGuid().ToString();
             Trace.TraceInformation("Host name {0}", hostName);
 
-            // monitored collection info 
-            var documentCollectionLocation = new DocumentCollectionInfo
+            var docTransformer = new DefaultDocumentTransformer();
+          
+            if (!String.IsNullOrEmpty(config.BlobConnectionString))
             {
-                Uri = new Uri(this.config.MonitoredUri),
-                MasterKey = this.config.MonitoredSecretKey,
-                DatabaseName = this.config.MonitoredDbName,
-                CollectionName = this.config.MonitoredCollectionName
-            };
+                BlobServiceClient blobServiceClient = new BlobServiceClient(config.BlobConnectionString);
+                containerClient = blobServiceClient.GetBlobContainerClient(config.BlobContainerName);
+                await containerClient.CreateIfNotExistsAsync();
+            }
 
-            var policy = new ConnectionPolicy()
+            DateTime starttime = DateTime.MinValue.ToUniversalTime();
+            if (config.DataAgeInHours.HasValue)
             {
-                ConnectionMode = ConnectionMode.Direct,
-                ConnectionProtocol = Protocol.Tcp
-            };
+                if (config.DataAgeInHours.Value >= 0)
+                {
+                    starttime = DateTime.UtcNow.Subtract(TimeSpan.FromHours(config.DataAgeInHours.Value));
+                }
+            }
 
-            policy.PreferredLocations.Add("North Europe");
+            changeFeedProcessor = sourceCollectionClient.GetContainer(config.MonitoredDbName, config.MonitoredCollectionName)
+                .GetChangeFeedProcessorBuilder<Document>("Live Data Migrator", ProcessChangesAsync)
+                .WithInstanceName("consoleHost")
+                .WithLeaseContainer(sourceCollectionClient.GetContainer(config.LeaseDbName, config.LeaseCollectionName))
```

Comment:
is this based on the assumption that the LeaseCollection and SourceCollection share the same account?

Reply:
Good spot, I have addressed this in last update.